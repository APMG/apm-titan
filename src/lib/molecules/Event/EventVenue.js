import React from 'react';
import PropTypes from 'prop-types';
import TagLink from '../../atoms/TagLink/TagLink';

const EventVenue = (props) => {
  return (
    // TODO: I don't think this should be an article. It seems like more of an attribute of a larger article, such as the event itself or eventInfo. The things it holds don't seem like paragraphs either.
    <article className="eventInfo-venue">
      <h2>Location</h2>
      <TagLink
        href={props.venue.website}
        title={props.venue.name}
        elementClass="link-plain"
      />
      <p>
        {props.venue.street} {props.venue.city}, {props.venue.state}{' '}
        {props.venue.zip}
      </p>
      <p>{props.venue.phone}</p>
      <p>
        {props.venue.latitude && props.venue.longitude && (
          <a
            href={`https://www.google.com/maps/?q=${props.venue.latitude},${
              props.venue.longitude
            }`}
            target="_blank"
            rel="noopener noreferrer"
            className="link"
          >
            View Map
          </a>
        )}
      </p>
      <br />
    </article>
  );
};

EventVenue.propTypes = {
  venue: PropTypes.shape({
    name: PropTypes.string,
    website: PropTypes.string,
    street: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    zip: PropTypes.string,
    phone: PropTypes.string,
    latitude: PropTypes.string,
    longitude: PropTypes.string
  })
};
export default EventVenue;
