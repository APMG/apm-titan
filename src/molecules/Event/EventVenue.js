import React from 'react';
import PropTypes from 'prop-types';

const EventVenue = (props) => {
  return (
    <div className="eventInfo-venue">
      <h2>Location</h2>
      <a
        href={props.venue.website}
        target="_blank"
        rel="noopener noreferrer"
        className="link-plain"
      >
        {props.venue.name}
      </a>
      <address>
        {props.venue.street} {props.venue.city}, {props.venue.state}
        <br />
        {props.venue.zip}
        <br />
        {props.venue.phone}
      </address>

      {props.venue.latitude && props.venue.longitude && (
        <a
          href={`https://www.google.com/maps/?q=${props.venue.latitude},${props.venue.longitude}`}
          target="_blank"
          rel="noopener noreferrer"
          className="link"
        >
          View Map
        </a>
      )}
    </div>
  );
};

EventVenue.propTypes = {
  venue: PropTypes.shape({
    name: PropTypes.string,
    website: PropTypes.string,
    street: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    zip: PropTypes.string,
    phone: PropTypes.string,
    latitude: PropTypes.string,
    longitude: PropTypes.string
  })
};
export default EventVenue;
