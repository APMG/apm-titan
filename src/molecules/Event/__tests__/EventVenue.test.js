import React from 'react';
import { render, cleanup } from '@testing-library/react';
import EventVenue from '../EventVenue';

// automatically unmount and cleanup DOM after the test is finished
afterEach(cleanup);

// Use this object for props that get reused
function defaultProps() {
  let venue = {
    name: 'First Avenue',
    website: 'https://youtube.com',
    street: '701 N 1st Ave',
    city: 'Minneapolis',
    state: 'MN',
    zip: '55403',
    phone: '(612) 338-8388',
    latitude: '44.9786311',
    longitude: '-93.2773028'
  };

  return {
    venue
  };
}

test('Creates the correct address element', () => {
  const { venue } = defaultProps();

  const { container } = render(<EventVenue venue={venue} />);

  const node = container.querySelectorAll('address')[0];

  // a snapshot works well here because 1) it's tiny and
  // 2) this should never change. if it does, there's a problem.
  expect(node).toMatchSnapshot();
});

test('Links to the correct Google Maps coordinates', () => {
  const { venue } = defaultProps();

  const { getByText } = render(<EventVenue venue={venue} />);

  const node = getByText('View Map');

  expect(node.getAttribute('href')).toBe(
    `https://www.google.com/maps/?q=${venue.latitude},${venue.longitude}`
  );
});
