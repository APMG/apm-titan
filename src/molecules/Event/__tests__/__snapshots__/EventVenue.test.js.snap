// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates the correct address element 1`] = `
<address>
  701 N 1st Ave
   
  Minneapolis
  , 
  MN
  <br />
  55403
  <br />
  (612) 338-8388
</address>
`;
